### PR TITLE
Remove zip code from duplicate patient lookup parameters

### LINKED ISSUE #3014: Adjust duplicate check endpoint to be First Name, Last Name, Facility, and Date of Birth
### WHY?

The current duplicate endpoint checks if a user is a duplicate based on their **first name, last name, DOB, and Zipcode**. Checking for zipcode creates more possibilities of missing duplicates due to:

- varying zipcode lengths (e.g. 94606 vs. 94606-2354 vs. 946062354)
- some users are highly transient, which means frequent changes to their Zipcode and the higher likelihood of users not receiving our duplicate prevention patterns.

By removing the zipcode endpoint, users are more likely to catch duplicate registrations sooner as they won't have to fill out the "contact information" section in order for our duplicate check to work.

**AS A** SR user

**I WANT** to be notified when I may be registering a duplicate patient

**SO THAT** I can stop the registration and in turn prevent duplicate patient profiles in my facility

---

**WHEN** the user enters the first name, last name, facility, and date of birth in the "add new person"/self-registration workflow

**THEN** run a duplicate check

---

Designs from previous story in case you need a reference point:
https://app.zenhub.com/workspaces/simplereport-better-data-team-605b43d51ea9f700119a48ee/issues/cdcgov/prime-simplereport/2318

### PR DESCRIPTION
**Pull request by nathancrtr**
## Related Issue or Background Info

- Closes #3014 
- **NOTE**: this is a re-release of a previously merged commit, with backwards compatibility concerns addressed in this iteration
    - See f428b37
    - Please also see demos below

## Changes Proposed
- Adds a new GraphQL query `patientExistsNew`; use query in frontend code
    - For backwards compatibility -- will be removed at some point in the future
- Backend resolvers & controllers for duplicate patient check do not expect zip code argument
- Backend service for duplicate patient check does not expect zip code argument
- Frontend removes zip code from identifying patient data tuple and is not used as a lookup parameter for an existing patient

## Additional Information
- N/A

### Add patient form
https://user-images.githubusercontent.com/27730981/146078902-e80ffcf6-cadb-43a9-8f5b-22348325b4d8.mov

### Self-registration
https://user-images.githubusercontent.com/27730981/146078922-d1f08fbe-0871-4309-9e40-aa65d3a3f659.mov

#### New UI
https://user-images.githubusercontent.com/27730981/146079036-3c5540d0-af95-4dbd-b13e-73bd8d7a35fc.mov

#### Stale UI (`git checkout main frontend/`)
https://user-images.githubusercontent.com/27730981/146079037-547c7483-f368-44df-a911-ec22a4c4275d.mov

## Checklist for Author and Reviewer
- [ ] Is this backwards compatible with the previous version of the UI?

### UI
- [x] Any changes to the UI/UX are approved by design 
- [x] Any new or updated content (e.g. error messages) are approved by design 

### Testing
- [x] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [x] Database changes are submitted as a separate PR
  - [x] Any new tables that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
  - [x] Any changes to tables that have custom no-PHI views are accompanied by changes to those views
        (including re-granting permission to the no-PHI user if need be)
  - [x] Liquibase rollback has been tested locally using `./gradlew liquibaseRollbackSQL` or `liquibaseRollback`
  - [x] Each new changeset has a corresponding [tag](https://docs.liquibase.com/change-types/community/tag-database.html)
- [x] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [x] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [x] Any dependencies introduced have been vetted and discussed

## Cloud
- [x] DevOps team has been notified if PR requires ops support
- [x] If there are changes that cannot be tested locally, this has been deployed to our Azure `test`, `dev`, or `pentest` environment for verification

### CODE REVIEW

File: backend/src/main/java/gov/cdc/usds/simplereport/api/patient/PatientResolver.java [hunk 1]
```diff
@@ -37,13 +37,25 @@ public long patientsCount(UUID facilityId, boolean showDeleted, String namePrefi
...
     Optional<Facility> facility =
         facilityId == null
             ? Optional.empty()
             : Optional.of(_os.getFacilityInCurrentOrg(facilityId));
 
-    return _ps.isDuplicatePatient(firstName, lastName, birthDate, zipCode, org, facility);
+    return _ps.isDuplicatePatient(firstName, lastName, birthDate, org, facility);
+  }
+
+  public boolean patientExistsNew(
```
Comment (emmastephenson) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/patient/PatientResolver.java:50]: Could this be renamed to patientExistsWithoutZip or similar?

**Review by emmastephenson — APPROVED** [review]
Overall LGTM, just one comment on naming.

File: backend/src/main/java/gov/cdc/usds/simplereport/api/patient/PatientResolver.java (hunk 1 above)
Comment (nathancrtr) [reply backend/src/main/java/gov/cdc/usds/simplereport/api/patient/PatientResolver.java]: Happy to! The `New` suffix has been a soft convention on the team when slotting in new queries alongside backwards-compatible ones, but it obviously doesn't carry any semantic information. See f49d522.

**Review by emmastephenson — APPROVED** [review]
LGTM!

File: backend/src/main/java/gov/cdc/usds/simplereport/api/patient/PatientResolver.java [hunk 2]
```diff
@@ -37,13 +37,25 @@ public long patientsCount(UUID facilityId, boolean showDeleted, String namePrefi
 
   public boolean patientExists(
       String firstName, String lastName, LocalDate birthDate, String zipCode, UUID facilityId) {
+    // Backwards compatibility shim -- zipCode is unused
```
Comment (nickclyde) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/patient/PatientResolver.java:40]: :+1:

**Review by nickclyde — APPROVED** [review]
Looks good! Thanks for maintaining compatibility :+1:

**Review by zdeveloper — APPROVED** [review]
LGTM! thanks for the updates

**Pull request merged**